src/IO/ADIOS/CommonADIOS1IOHandler.cpp

```diff
@@ -533,32 +568,89 @@ CommonADIOS1IOHandlerImpl::openDataset(Writable* writable,
     VERIFY(vi != nullptr, "Internal error: Failed to inquire about ADIOS variable during dataset opening");
 
     Datatype dtype;
+
+    // note the ill-named fixed-byte adios_... types
+    // https://github.com/ornladios/ADIOS/issues/187
     switch( vi->type )
     {
         using DT = Datatype;
         case adios_byte:
             dtype = DT::CHAR;
             break;
         case adios_short:
-            dtype = DT::INT16;
+            if( sizeof(short) == 2u )
+                dtype = DT::SHORT;
+            else if( sizeof(int) == 2u )
+                dtype = DT::INT;
+            else if( sizeof(long) == 2u )
+                dtype = DT::LONG;
+            else if( sizeof(long long) == 2u )
+                dtype = DT::LONGLONG;
+            else
+                throw unsupported_data_error("No native equivalent for Datatype adios_short found.");
```

Comment:
`long` is at least 32 Bit.
`long long` is at least 64 Bit.
Which maskes the 
```cpp
else if( sizeof(long) == 2u )
else if( sizeof(long long) == 2u )
```
branches dead code.
[C++ fundamental types](https://en.cppreference.com/w/cpp/language/types)